### ISSUE TITLE
About the statistics of the dataset

### ISSUE THREAD

**Issue by David-Lee-1990** [Babelscape/rebel#16]
Hi, nice work. I try to replica the pretraining process, but using "[rebel](https://github.com/Babelscape/rebel)/datasets/rebel-short.py /", i can not get the same train/val/test num writen in the paper as 
![image](https://user-images.githubusercontent.com/29653610/165078932-90b7f12b-8e62-41c6-9218-5d850c5cb6e7.png)

plz enlight me some details!±±

**Comment by David-Lee-1990** [comment]
Hi, on test set, i get 172,007 instances and 451,623 triples. 

`import json
from collections import Counter, defaultdict
import re
from tqdm import tqdm
import pandas as pd
relations_df = pd.read_csv('./data/rebel/relations_count.tsv', header = None, sep='\t')
relations = list(relations_df[0])

def rebel2conll04(case):
    filepath = './data/rebel/en_%s.jsonl'%case
    with open(filepath, encoding="utf-8") as f:
        useful_instance_count = 0
        useful_triple_count = 0
        for row in tqdm(f.readlines()):
            article = json.loads(row)
            prev_len = 0
            if len(article['triples']) == 0:
                continue
            for text_paragraph in article['text'].split('\n'):
                if len(text_paragraph) == 0:
                    continue
                sentences = re.split(r'(?<=[.])\s', text_paragraph)
                text = ''
                for sentence in sentences:
                    text += sentence + ' '
                    if any([entity['boundaries'][0] < len(text) + prev_len < entity['boundaries'][1] for entity in article['entities']]):
                        continue
                    entities = sorted([entity for entity in article['entities'] 
                                       if prev_len < entity['boundaries'][1] <= len(text)+prev_len], key=lambda tup: tup['boundaries'][0])
                    decoder_output = '<triplet> '
                    for int_ent, entity in enumerate(entities):
                        triplets = sorted([triplet for triplet in article['triples'] 
                                           if triplet['subject'] == entity and prev_len< triplet['subject']['boundaries'][1]<=len(text) + prev_len 
                                           and prev_len< triplet['object']['boundaries'][1]<=len(text)+ prev_len 
                                           and triplet['predicate']['surfaceform'] in relations], key=lambda tup: tup['object']['boundaries'][0])
                        if len(triplets) == 0:
                            continue
                        decoder_output += entity['surfaceform'] + ' <subj> '
                        for triplet in triplets:
                            decoder_output += triplet['object']['surfaceform'] + ' <obj> '  + triplet['predicate']['surfaceform'] + ' <subj> '
                            useful_triple_count += 1
                        decoder_output = decoder_output[:-len(' <subj> ')]
                        decoder_output += ' <triplet> '
                    decoder_output = decoder_output[:-len(' <triplet> ')]
                    prev_len += len(text)

                    if len(decoder_output) == 0:
                        text = ''
                        continue
                    useful_instance_count += 1
                    text = ''
    
    print(useful_instance_count, useful_triple_count)


def get_relation_statistics():
    file = './data/rebel_pretrain/en_train.jsonl'
    relation_dict = defaultdict(int)
    with open(file, 'r') as f:
        lines = f.readlines()
        for line in tqdm(lines):
            article = json.loads(line)
            for triplet in article['triples']:
                relation_dict[triplet['predicate']['surfaceform']] += 1
    sorted_relations = sorted([(k,v) for (k,v) in relation_dict.items()], key=lambda x:x[1], reverse=True)
    print(len(sorted_relations))
    with open('./data/rebel_pretrain/filtered_relations_total.txt', 'w') as f:
        for r in sorted_relations:
            f.write(f'{r[0]}\t{r[1]}\n')
    with open('./data/rebel_pretrain/filtered_relations_220.txt', 'w') as f:
        for r in sorted_relations[:220]:
            f.write(f'{r[0]}\t{r[1]}\n')
    
    
if __name__ == "__main__":
    rebel2conll04(case='test')`

**Comment by LittlePea13** [comment]
Hi, sorry I haven't get back at you about this, I want to check myself but haven't found the time. 

In order not to make you wait more, I think it is due to the fact I used a % of those files as validation and test set (which are the numbers reported in the paper). However this data was not supervised, and I wouldn't use those splits for evaluation, at least not as they are. If you are using the dataset for training, I suggest you use a small portion of the validation data in order to do early stopping and check during training.
 
I will get back at you once I check myself the data.

**Comment by debraj135** [comment]
It would be great if you could explain how the number of instances for REBEL (full) is 2,754,387 whereas that for REBEL (sent.) is 784,202. 

Am I correct in interpreting an instance as being a unique sentence?

In particular, could you please elaborate on what kind of processing was done on REBEL (full) to get to REBEL (sent.). 

I think the REBEL dataset [here](https://huggingface.co/datasets/Babelscape/rebel-dataset) has all over 3 million instances in the training set and it is not clear to me how to get to 784,202 instances from here.

**Comment by LittlePea13** [comment]
Hi @debraj135, (sent.) is for a sentence level instance, hence the paragraphs (wikipedia abstracts) are split into sentences. However the number in the paper (784,202) is for instances that contain one of the 220 relation types selected to pretrain rebel, hence if you use the dataset [here](https://huggingface.co/datasets/Babelscape/rebel-dataset) but only keep those from the 220 relation types as in here https://github.com/Babelscape/rebel/blob/837983b7347bd65e0bf5ba59f8158797dd54d77f/datasets/rebel-short.py#L102-L103, then you should obtain the numbers in the paper.

**Comment by LittlePea13** [comment]
I can confirm that the numbers in the paper are for 10% of the validation and test files, since those were the numbers used for early stopping and reporting on performance for the silver data. If you need to replicate the same splits try setting val_percent_check to 0.1 and the seed to 42.

Nevertheless, while they were useful for the pre-training of REBEL, as mentioned before, they are quite noisy and unbalanced. To assess final performance is better to use a gold-standard dataset as the ones it was finetuned on.
